aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,151 @@
+
+## Code sharing and .NET Standard
+
+The [.NET Standard](/dotnet/standard/net-standard) is a formal specification of .NET APIs that are intended to be available on all .NET implementations. Mono on WebAssembly supports `netstandard2.0` or higher. .NET Standard class libraries can be shared across server code and in browser-based apps.
+
+Browsers support the APIs that developers use to build web apps. Not all .NET APIs are callable from the browser. For example, arbitrary TCP sockets can't be accessed in a browser, so [System.Net.Sockets.TcpListener](/dotnet/api/system.net.sockets.tcplistener) can't perform any useful task. For BCL APIs that don't apply to a given platform, the BCL throws a [PlatformNotSupported](/dotnet/api/system.platformnotsupportedexception) exception.
+
+## JavaScript/TypeScript interop
+
+For apps that require third-party JavaScript libraries and browser APIs, WebAssembly is designed to interoperate with JavaScript. Blazor is capable of using any library or API that JavaScript is able to use. The Mono team is working on a library that exposes standard browser APIs to .NET.
+
+## Optimization
+
+Traditionally, .NET has focused on platforms where the app's binary size isn't a major concern. It doesn't really matter whether a server-side ASP.NET app is 1MB or 50MB. It's only a moderate concern for native desktop or mobile apps. But for browser apps, payload size is critical.
+
+Development efforts are aimed at reducing the download size of the Mono runtime and .NET app assemblies. Here are three phases of size optimization the Blazor engineering team has in mind:
+
+1. Mono runtime stripping
+
+   The Mono runtime contains many desktop-specific features. We hope that the Blazor packages will contain a trimmed version of Mono that is substantially smaller than the full-fat distribution. In an optimization experiment, the Mono runtime was pruned of unnecessary code. Over 70% of the Mono *wasm* file was removed while keeping a basic app working.
+
+1. Publish-time IL stripping
+
+   The .NET IL linker (originally based on the Mono linker) performs static analysis to determine which parts of .NET assemblies can ever get called by an app, then it strips out everything else.
+
+   This is equivalent to *tree shaking* in JavaScript, except the IL linker is much more fine-grained, operating at the level of individual methods. The IL Linker removes all of the system library code that the app isn't using, which often results in a 70% or greater reduction in code size.
+
+1. Compression
+
+   Most web servers support HTTP compression, which typically cuts the remaining payload size by a further 75%.
+
+Overall, a .NET-based browser app is never going to be as tiny as a minimal React app, but the goal is to make it small enough that a typical user with average Internet bandwidth won't notice or care about an app's first load time. After first load, the app's assemblies are fully cached.
+
+## Deployment
```

Comment:
Blazor is targeted at all web developers, not just ASP.NET devs. Instead of focusing on the type of developer, we should say that you have the option of using Blazor for just client-side development, or for full-stack .NET development. The later has many advantages: same tooling, same build infrastructure, same language, ability to share code, etc. Full-stack support is relevant whether you are an existing ASP.NET dev or not.

Server-side rendering is not yet available, so let's leave that out. We